### ISSUE TITLE
tqdm rounds up and reports 100% before actually finishing

### ISSUE THREAD

**Issue by mlubej** [tqdm/tqdm#1398]
If there are enough steps in the loop so that the increment is small enough, it can happen that `tqdm` shows `100%` even though the process hasn't finished yet. This is the default behavior and I think it should be changed to stop at 99% before actually finishing.

MVE:

```python
from time import sleep
from tqdm.cli import tqdm

n = 1000
for idx in tqdm(range(n)):
    if idx == n-1:
        break
    sleep(1/n)
```

<img width="481" alt="image" src="https://user-images.githubusercontent.com/37869250/204011511-c8ed015d-aeea-4382-a0ec-01b690608b06.png">

I know it's possible to change the bar format to show decimals, e.g.

```python
from time import sleep
from tqdm.cli import tqdm

n = 1000
for idx in tqdm(range(n), bar_format = "{desc}: {percentage:.1f}%|{bar}| {n_fmt}/{total_fmt} [{elapsed}<{remaining}]"):
    if idx == n-1:
        break
    sleep(1/n)
```

<img width="371" alt="image" src="https://user-images.githubusercontent.com/37869250/204011558-ff7f10af-1434-4119-9654-426d217eb736.png">

but this probably isn't an acceptable solution for everyone.

**Comment by taoroalin** [comment]
Yes please stop rounding up to 100%! Either rounding down between 99 and 100, or showing decimals, would be awesome!

**Comment by carlfischerjba** [comment]
There's another PR for this: https://github.com/tqdm/tqdm/pull/1727
